tests/unit/features/StringFeatures_unittest.cc

```diff
@@ -104,3 +106,15 @@ TEST(StringFeaturesTest,clone)
 	SG_UNREF(f);
 	SG_UNREF(f_clone);
 }
+
+TEST(StringFeaturesTest,equals)
+{
+	SGStringList<char> strings = generateRandomStringData();
```

Comment:
we would want a test for `SGStringList` ... all shogun objects are tested for clone and equals in [here](https://github.com/shogun-toolbox/shogun/blob/develop/tests/unit/base/SGObjectAll_unittest.cc#L222)

Reply:
I've seen those. These are tests for constructors without arguments, right?

Reply:
For now, yes
We might add a version with random content soon

Reply:
Same here (remove this). I would however love to see a test for clone/equals of `SGStringList` as those are very low level operations

Reply:
And we should probably do these tests for all legal template arguments using the googletest magic.
@gf712 can help here

Reply:
Yup, for this you need to define the types you want to test:
https://github.com/shogun-toolbox/shogun/blob/13961a1d353414f301370425e124c18a4ce4c3b2/tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc#L47-L50
Associate it to a (empty) class:
https://github.com/shogun-toolbox/shogun/blob/13961a1d353414f301370425e124c18a4ce4c3b2/tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc#L26-L29
https://github.com/shogun-toolbox/shogun/blob/13961a1d353414f301370425e124c18a4ce4c3b2/tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc#L59
And then just pass the class to the TYPED_TEST macro
https://github.com/shogun-toolbox/shogun/blob/13961a1d353414f301370425e124c18a4ce4c3b2/tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc#L64
In the test you can access the type with `TypeParam`